planner/core/find_best_task.go

```diff
@@ -407,17 +420,163 @@ func (ds *DataSource) findBestTask(prop *property.PhysicalProperty) (t task, err
 			}
 			continue
 		}
-		idxTask, err := ds.convertToIndexScan(prop, candidate)
+		if !isIdxMergePath {
+			idxTask, err := ds.convertToIndexScan(prop, candidate)
+			if err != nil {
+				return nil, err
+			}
+			if idxTask.cost() < t.cost() {
+				t = idxTask
+			}
+			continue
+		}
+		idxMergeTask, err := ds.convertToIndexMergeScan(prop, candidate)
 		if err != nil {
 			return nil, err
 		}
-		if idxTask.cost() < t.cost() {
-			t = idxTask
+		if idxMergeTask.cost() < t.cost() {
+			t = idxMergeTask
 		}
 	}
 	return
 }
 
+func (ds *DataSource) convertToIndexMergeScan(prop *property.PhysicalProperty, candidate *candidatePath) (task task, err error) {
+	if prop.TaskTp == property.CopSingleReadTaskType {
+		return invalidTask, nil
+	}
+	if !prop.IsEmpty() && !candidate.isMatchProp {
+		return invalidTask, nil
+	}
+	path := candidate.path
+	totalRowCount := 0.0
+	partPaths := make([]PhysicalPlan, 0, len(path.partialIndexPaths))
+	cop := &copTask{}
+	for _, partPath := range path.partialIndexPaths {
+		if partPath.isTablePath {
```

Comment:
Of course, we need a `PhysicalTableScan` because `IndexMergePath` must be double read. `partPath`, there, is used to get `rowid` for the conditions with primary key.